### PR TITLE
EIP-712 wallet linking changes to the web3 package

### PR DESCRIPTION
**Pull request by tak-hntlabs**
branched off G's wallet-link changes for the smart contract. https://github.com/river-build/river/pull/189

### CODE REVIEW

**Review by giuseppecrj — APPROVED** [review]

File: packages/web3/src/v3/WalletLink.ts [hunk 1]
```diff
@@ -51,22 +60,32 @@ export class WalletLink {
...
+        const { domain, types, value } = createEip712LinkedWalletdData({
+            domain: this.eip712Domain,
+            nonce: rootKeyNonce,
+            wallet: walletAddress,
+        })
+        return this.signTypedData(rootKey, domain, types, value)
     }
 
-    private generateWalletSignatureForRootKey({
+    private async generateWalletSignatureForRootKey({
```
Comment (salzbrenner) [inline packages/web3/src/v3/WalletLink.ts:74]: why async?

**Review by salzbrenner — APPROVED** [review]
🔥

File: packages/web3/src/v3/WalletLink.ts (hunk 1 above)
Comment (tak-hntlabs) [reply packages/web3/src/v3/WalletLink.ts]: good eye. Had an await in this method before I refactored code. Removed.

**Comment by tak-hntlabs** [comment]
merged this change into G's branch to get CI to pass. Not needed anymore. Close.

**Pull request merged**